app/lib/sharezone_plus/page/sharezone_plus_page_controller.dart

```diff
@@ -101,6 +105,93 @@ class SharezonePlusPageBloc extends Bloc<PlusPageEvent, PlusPageViewModel> {
   }
 }
 
+@riverpod
+class SharezonePlusPageNotifier extends _$SharezonePlusPageNotifier {
+  late RevenueCatPurchaseService _purchaseService;
+  late SubscriptionService _subscriptionService;
+
+  SharezonePlusPageNotifier() {
+    // Fake loading time (for development)
+    Future.delayed(const Duration(seconds: 1), () async {
+      _subscriptionService.isSubscriptionActiveStream().listen((hasPlus) {
+        add(_PlusStatusChanged(hasPlus: hasPlus));
+      });
+    });
+  }
+
+  @override
+  PlusPageViewModel build(BuildContext context) {
+    _purchaseService = context.read<RevenueCatPurchaseService>();
+    _subscriptionService = context.read<SubscriptionService>();
+    return PlusPageLoading();
```

Comment:
Normally, you wouldn't pass the `BuildContext`. I just did it because the other classes are not migrated to `riverpod`.